packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -111,12 +111,6 @@ Future<XcodeBuildResult> buildXcodeProject({
   await removeFinderExtendedAttributes(app.project.hostAppRoot, processUtils, globals.logger);
 
   final XcodeProjectInfo projectInfo = await globals.xcodeProjectInterpreter.getInfo(app.project.hostAppRoot.path);
-  if (!projectInfo.targets.contains('Runner')) {
-    globals.printError('The Xcode project does not define target "Runner" which is needed by Flutter tooling.');
-    globals.printError('Open Xcode to fix the problem:');
-    globals.printError('  open ios/Runner.xcworkspace');
-    return XcodeBuildResult(success: false);
-  }
```

Comment:
Important change 1: remove the target check

Reply:
Thanks for the walkthrough :)